h2o-core/src/test/java/water/fvec/TestFrameBuilder.java

```diff
@@ -412,6 +416,13 @@ private void checkColumnData() {
             throwIf(numRows != stringData.get(colIdx).length, "Columns have different number of elements");
           }
           break;
+        case Vec.T_BAD:
+          final double[] data = numericData.get(colIdx);
+          numRows = data.length;
+          for (int i = 0; i < data.length; i++) {
+            throwIf(!Double.isNaN(data[i]), "All elements in a bad column must be NAs or zeros");
```

Comment:
zeros? why would a constant 0 chunk be BAD?

Reply:
Good point, I even had it in the error message :) Support added in https://github.com/h2oai/h2o-3/pull/3565/commits/2838b9968189eec8bbba95e1c4eb4244969bce60. 

Had to tell NewChunk to consider all 0 chunks as BAD as well.

Reply:
Correction, addressed in https://github.com/h2oai/h2o-3/commit/58445e683c0a3be2dea1d4d37c97403cba49f580

Reply:
Makes sense, but `NewChunk` consideres BAD types only a vec full of NAs. Which also makes sense. It's a valid chunk full of constant observations.

```
    if(_naCnt == _len)          // All NAs ==> NA Chunk
      return Vec.T_BAD;
```

Reply:
Considering a chunk full of zeroes to be `BAD` breaks most of our tests. Setting it to be BAD only if the VEC is pre-assigned as BAD (and not null) is also possible, but it makes no sense to me. I'd stick with current solution, as we're able to represent BAD chunks in tests and deal with this separately.

Reply:
@Pscheidl I was referring to your error message "All elements in a bad column must be NAs or zeros"

I don't understand the mention of zeros